### PR TITLE
Re-enable codesign for maven packages

### PR DESCRIPTION
**Pull request by snnn**
### Description
PR #22217 was reverted.  This PR re-enable it.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+parameters:
+  - name: JarFileDirectory
+    type: string
+
+steps:
+  - task: AzureKeyVault@2
+    displayName: 'Get GnuPG signing keys'
+    inputs:
+      azureSubscription: 'OnnxrunTimeCodeSign_20240611'
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:9]: NIT: Maybe make a comment here and showing this is a Service Connection under Lotus? This way people will not go searching for an Azure Subscription with this name.

File: tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        jar_file_directory='${{ parameters.JarFileDirectory }}'
+        working_directory='$(Build.SourcesDirectory)'
+        original_private_key='$(java-pgp-key)'
+        original_passphrase='$(java-pgp-pwd)'
+
+        private_key_file=$working_directory/private_key.txt
+        passphrase_file=$working_directory/passphrase.txt
+
+        echo "Generating GnuPG key files."
+        printf "%s" "$original_private_key" >$private_key_file
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:31]: Do we still want to print the keys.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:31]: They are not visible from build logs.

Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:31]: This dest of this printf is a file.

File: tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      SecretsFilter: 'java-pgp-pwd,java-pgp-key'
+      RunAsPreJob: false
+
+  - task: CmdLine@2
+    displayName: 'Sign jar files: GnuPG and sha256'
+    inputs:
+      workingDirectory: '$(Build.SourcesDirectory)'
+      script: |
+        #!/bin/bash
+        set -x
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:20]: Would `set -x` enable debug mode and show the variables that being used. For example, $original_private_key

File: tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:9]: The value for this property can only be a Service Connection Name.  We cannot directly use an Azure subscription in pipelines. We must create a service connection first.

**Comment by jchen351** [comment]
Can you rerun/fix the pancaking pipeline?

File: tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml (hunk 3 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/jar-maven-signing-linux.yml:20]: Should be e. Sorry.

**Comment by snnn** [comment]
> Can you rerun/fix the packaging pipeline?

Would be no change. The pipeline is still broken due to a recent PR for adding multi LORA support.

**Review by jchen351 — APPROVED** [review]

**Pull request merged**